doc/storage/api/api.rst

```diff
@@ -140,6 +148,24 @@ These definitions must be defined in the header file :file:`psa/internal_trusted
 
     It will be incremented in small updates that are unlikely to include breaking changes.
 
+.. macro:: PSA_ITS_SUPPORT_ITERATION
+    (1u << 2)
+
+    Flag indicating that `psa_its_iterator_start()` and  `psa_its_iterator_next()` are supported.
+
+.. struct:: psa_its_storage_iterator_t
```

Comment:
I've changed my mind: as this is a fully implementation-defined type, we should probably use the same pattern as the Crypto API for such types, and just use a typedef in this specification. For example, see [the definition of `psa_key_attributes_t`](https://arm-software.github.io/psa-api/crypto/1.2/api/keys/attributes.html#c.psa_key_attributes_t). This gives the implementation full flexibility in the definition of the type.

To do this, replace:

```
.. struct:: psa_its_storage_iterator_t
    :type:
```

with

```
.. typedef:: /* implementation-defined type */ psa_its_storage_iterator_t
```

Likewise for the PS iterator type.

In the Crypto API, we then refer to these as 'objects' rather than 'structures' - i.e. use 'iterator object' in this spec when describing a variable of this type.